cli/src/git_util.rs

```diff
@@ -73,12 +73,72 @@ pub fn get_git_repo(store: &Store) -> Result<git2::Repository, CommandError> {
     }
 }
 
-pub fn is_colocated_git_workspace(_workspace: &Workspace, repo: &ReadonlyRepo) -> bool {
+pub fn is_colocated_git_workspace(
+    ui: Option<&Ui>,
+    workspace: &Workspace,
+    repo: &ReadonlyRepo,
+) -> bool {
     let Some(git_backend) = repo.store().backend_impl().downcast_ref::<GitBackend>() else {
         return false;
     };
 
-    git_backend.is_colocated()
+    if git_backend.is_colocated() {
+        return true;
+    }
+
+    if let Some(ui) = ui {
+        let workspace_dot_git = workspace.workspace_root().join(".git");
+        let Ok(meta) = workspace_dot_git.symlink_metadata() else {
+            return git_backend.is_colocated();
+        };
+        // Check symlink first as is_dir follows symlinks
+        if meta.is_symlink() {
+            let target = std::fs::read_link(&workspace_dot_git)
+                .unwrap_or_else(|_| Path::new("<could not read link>").to_path_buf());
+            writeln!(
+                ui.warning_default(),
+                "Workspace has a .git symlink (pointing to {}), that isn't pointing to JJ's git \
+                 repo",
+                target.display()
+            )
+            .ok();
+        } else if meta.is_dir() {
+            writeln!(
+                ui.warning_default(),
+                "Workspace has a .git directory that is not managed by JJ"
+            )
+            .ok();
+        } else if let Ok(_worktree_repo) = gix::open(workspace.workspace_root()) {
```

Comment:
True, I had it like this because there are more details you can add to the warning if you don't have to make the tests work on windows with its myriad path formats, and I had to remove them.